sycl/doc/extensions/proposed/sycl_ext_oneapi_named_sub_group_sizes.asciidoc

```diff
@@ -0,0 +1,301 @@
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2019-2022 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties]
+
+* link:../proposed/sycl_ext_oneapi_kernel_properties.asciidoc[
+  sycl_ext_oneapi_kernel_properties]
+
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+
+== Overview
+
+SYCL provides a mechanism to set a required sub-group size for a kernel via
+an attribute, and the sycl_ext_oneapi_kernel_properties extension provides an
+equivalent property.
+
+Either mechanism is sufficient when tuning individual kernels for specific
+devices, but their usage quickly becomes complicated in real-life scenarios
+because:
+
+1. An integral sub-group size must be provided at host compile-time.
+
+2. The sub-group sizes supported by a device are not known until run-time.
+
+3. It is common for the same sub-group size to be used for all kernels
+   (e.g. because the sub-group size is reflected in data structures).
+
+Applications wishing to write portable sub-group code that can target multiple
+architectures must therefore multi-version their C++ code (e.g. via templates),
+dispatch to the correct kernel(s) based on the result of a run-time query, and
+repeat this process for every kernel individually.
+
+This extension aims to simplify the process of using sub-groups by introducing
+the notion of _named_ sub-group sizes, allowing developers to request a
+sub-group size that meets certain requirements at host compile-time and
+deferring the selection of a specific sub-group size until the kernel is
+compiled for a specific device.
+
+This extension also changes the default behavior of sub-groups in SYCL code to
```

Comment:
```suggestion
This extension also defines the default behavior of sub-groups in SYCL code to
```

Maybe this is better?  The default behavior is unspecified in the core SYCL spec, so the extension doesn't necessarily change it.